support/oidc-discovery-provider/domain_policy_test.go

```diff
@@ -0,0 +1,40 @@
+package main
+
+import (
+	"testing"
+
+	"github.com/stretchr/testify/assert"
+	"github.com/stretchr/testify/require"
+)
+
+func TestDomainAllowlist(t *testing.T) {
+	t.Run("unicode", func(t *testing.T) {
+		policy, err := DomainAllowlist("😬.test")
+		require.NoError(t, err)
+		assert.NoError(t, policy("😬.test"))
+		assert.NoError(t, policy("xn--n38h.test"))
+		assert.EqualError(t, policy("bad.test"), `domain "bad.test" is not allowed`)
+	})
+
+	t.Run("punycode", func(t *testing.T) {
+		policy, err := DomainAllowlist("xn--n38h.test")
+		require.NoError(t, err)
+		assert.NoError(t, policy("😬.test"))
+		assert.NoError(t, policy("xn--n38h.test"))
+		assert.EqualError(t, policy("bad.test"), `domain "bad.test" is not allowed`)
+	})
+
+	t.Run("ascii", func(t *testing.T) {
+		policy, err := DomainAllowlist("ascii.test")
+		require.NoError(t, err)
+		assert.NoError(t, policy("ascii.test"))
+		assert.EqualError(t, policy("bad.test"), `domain "bad.test" is not allowed`)
+	})
+}
+
+func TestAllowAnyDomain(t *testing.T) {
```

Comment:
is it possible that an "invalid" domain came here? I'm asking about for example something like: "invalid\\domain.com", if that is the case, may we at least validate domain format?

Reply:
That's probably reasonable.